proxy/coco/src/state.rs

```diff
@@ -328,12 +347,12 @@ impl State {
     /// * If no project for the `urn` was found.
     /// * If the [`git::Browser`] fails.
     /// * If the passed `callback` errors.
-    pub fn with_browser<F, T>(&self, urn: &RadUrn, callback: F) -> Result<T, Error>
+    pub async fn with_browser<F, T>(&self, urn: RadUrn, callback: F) -> Result<T, Error>
```

Comment:
Might be worth considering to maintain a pool of `Browser`s here. Also, `spawn_blocking` the callback seems advisable.

Reply:
#973